webapp/src/main/java/org/apache/atlas/web/rest/EntityREST.java

```diff
@@ -1130,40 +1131,42 @@ public EntityAuditSearchResult searchAuditEvents(AuditSearchParams parameters) t
 
             EntityAuditSearchResult ret = esBasedAuditRepository.searchEvents(dslString);
 
-            scrubEntityAudits(ret);
+            scrubEntityAudits(ret, parameters.getSuppressLogs());
 
             return ret;
         } finally {
             AtlasPerfTracer.log(perf);
         }
     }
 
-    private void scrubEntityAudits(EntityAuditSearchResult result) throws AtlasBaseException {
+    private void scrubEntityAudits(EntityAuditSearchResult result, boolean suppressLogs) throws AtlasBaseException {
         AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("scrubEntityAudits");
         for (EntityAuditEventV2 event : result.getEntityAudits()) {
             try {
                 AtlasEntityWithExtInfo entityWithExtInfo = entitiesStore.getByIdWithoutAuthorization(event.getEntityId());
                 AtlasEntityHeader entityHeader = new AtlasEntityHeader(entityWithExtInfo.getEntity());
-
-                AtlasAuthorizationUtils.verifyAccess(new AtlasEntityAccessRequest(typeRegistry, ENTITY_READ, entityHeader), "read entity audit: guid=", event.getEntityId());
+                AtlasSearchResult ret = new AtlasSearchResult();
+                ret.addEntity(entityHeader);
+                AtlasSearchResultScrubRequest request = new AtlasSearchResultScrubRequest(typeRegistry, ret);
+                AtlasAuthorizationUtils.scrubSearchResults(request, suppressLogs);
```

Comment:
Instead of calling AtlasAuthorizationUtils.scrubSearchResults for each event, just collects list of all entityHeaders & then scrub all of them at once with single call to AtlasAuthorizationUtils.scrubSearchResults()

Reply:
@nikhilbonte21 
Overall complexity in both the approach is same, the approach you are suggesting will require parallel map of entity and its authz status. then lookup for the authz status and set  event.setDetail(null) accordingly.
 therefore I propose the above approach. LMK if you have concerns on it